glib/tests/fileutils.c

```diff
@@ -860,10 +860,19 @@ test_stdio_wrappers (void)
 
   cwd = g_get_current_dir ();
   path = g_build_filename (cwd, "mkdir-test", NULL);
-  g_free (cwd);
   ret = g_chdir (path);
-  g_assert (errno == EACCES);
-  g_assert (ret == -1);
+  if (geteuid () == 0)
```

Comment:
This looks fixed upstream as well
https://git.gnome.org/browse/glib/commit/?id=202d7d37d6ce066291504276810c350c9a3feb7a